lib/daftka/metadata/store.ex

```
@@ -42,7 +42,40 @@ defmodule Daftka.Metadata.Store do
   @spec start_link(keyword()) :: Agent.on_start()
   def start_link(opts \\ []) do
     initial_state = %State{topics: %{}}
-    Agent.start_link(fn -> initial_state end, Keyword.merge([name: __MODULE__], opts))
+    default_name = server_name()
+    name = Keyword.get(opts, :name, default_name)
+
+    case Agent.start_link(fn -> initial_state end, Keyword.merge([name: name], opts)) do
+      {:ok, pid} = ok ->
+        # Also register a local alias for compatibility with tests
+        try do
```

doing Process.register for everybody is a good idea. but dont mask errors.